pkg/dynamicconfig/reflectors.go

```diff
@@ -100,15 +101,26 @@ func configMapReflector(name string, namespace string, cs k8sClient.Clientset, t
 	return k8s.ReflectorConfig[DynamicConfig]{
 		Name:  "cm-" + name + "-" + namespace,
 		Table: t,
-		TransformMany: func(_ statedb.ReadTxn, o any) []DynamicConfig {
-			cm := o.(*v1.ConfigMap).DeepCopy()
```

Comment:
I removed all the `DeepCopy()` calls as I nowhere saw these original objects being mutated, and even if they would've been nothing else has access to these so it doesn't even matter if they did get mutated. This `o` object is only for use by `TransformMany` after which it is thrown away.